### PR TITLE
[FLINK-34916][table] Support `ALTER CATALOG SET` syntax

### PR DESCRIPTION
**Pull request by liyubin117**
## What is the purpose of the change

Set one or more properties in the specified catalog. If a particular property is already set in the catalog, override the old value with the new one.

## Brief change log

* ALTER CATALOG catalog_name SET (key1=val1, ...)

## Verifying this change

This change added tests and can be verified as follows:

flink-table/flink-sql-client/src/test/resources/sql/catalog_database.q
flink-table/flink-sql-gateway/src/test/resources/sql/catalog_database.q

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? yes

### CODE REVIEW

**Comment by liyubin117** [comment]
@flinkbot run azure

**Comment by liyubin117** [comment]
@flinkbot run azure

**Comment by liyubin117** [comment]
@flinkbot run azure

**Comment by liyubin117** [comment]
@LadyForest Hi, CI passed, looking forward your review, thanks!

**Review by LadyForest — COMMENTED** [review]
Thanks for the contribution! I only left some minor comments. PTAL

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java [hunk 1]
```diff
@@ -321,6 +322,42 @@ public void createCatalog(String catalogName, CatalogDescriptor catalogDescripto
...
+                "Catalog name cannot be null or empty.");
+        checkNotNull(catalogDescriptor, "Catalog descriptor cannot be null");
+        CatalogStore catalogStore = catalogStoreHolder.catalogStore();
+        Optional<CatalogDescriptor> oldCatalogDescriptor = getCatalogDescriptor(catalogName);
+        if (catalogStore.contains(catalogName) && oldCatalogDescriptor.isPresent()) {
+            Map<String, String> props = oldCatalogDescriptor.get().getConfiguration().toMap();
+            props.putAll(catalogDescriptor.getConfiguration().toMap());
+            CatalogDescriptor newCatalogDescriptor =
+                    CatalogDescriptor.of(catalogName, Configuration.fromMap(props));
+            Catalog catalog = initCatalog(catalogName, newCatalogDescriptor);
```
[the remark this answers is not in the document]
Comment (liyubin117) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java:347]: Good idea :)

File: flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl [hunk 2]
```diff
@@ -147,6 +147,28 @@ SqlDrop SqlDropCatalog(Span s, boolean replace) :
     }
 }
 
+/**
+* Parses a alter catalog statement.
```
[the remark this answers is not in the document]
Comment (liyubin117) [reply flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl]: done

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlDdlToOperationConverterTest.java [hunk 3]
```diff
@@ -102,6 +103,23 @@
...
+        assertThat(operation).isInstanceOf(AlterCatalogOptionsOperation.class);
+        assertThat(((AlterCatalogOptionsOperation) operation).getCatalogName()).isEqualTo("cat2");
+        assertThat(operation.asSummaryString())
+                .isEqualTo("ALTER CATALOG cat2\n  SET 'K1' = 'V1',\n  SET 'k2' = 'v2_new'");
+
+        final Map<String, String> expectedOptions = new HashMap<>();
+        expectedOptions.put("K1", "V1");
+        expectedOptions.put("k2", "v2_new");
+        assertThat(((AlterCatalogOptionsOperation) operation).getProperties())
+                .isEqualTo(expectedOptions);
```
[the remark this answers is not in the document]
Comment (liyubin117) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlDdlToOperationConverterTest.java]: Got it ! it is more concise now :)

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java [hunk 4]
```diff
@@ -321,6 +322,42 @@ public void createCatalog(String catalogName, CatalogDescriptor catalogDescripto
...
+    public void alterCatalog(String catalogName, CatalogDescriptor catalogDescriptor)
+            throws CatalogException {
+        checkArgument(
+                !StringUtils.isNullOrWhitespaceOnly(catalogName),
+                "Catalog name cannot be null or empty.");
+        checkNotNull(catalogDescriptor, "Catalog descriptor cannot be null");
+        CatalogStore catalogStore = catalogStoreHolder.catalogStore();
+        Optional<CatalogDescriptor> oldCatalogDescriptor = getCatalogDescriptor(catalogName);
+        if (catalogStore.contains(catalogName) && oldCatalogDescriptor.isPresent()) {
+            Map<String, String> props = oldCatalogDescriptor.get().getConfiguration().toMap();
```
[the remark this answers is not in the document]
Comment (liyubin117) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java]: Good idea

**Comment by liyubin117** [comment]
@LadyForest Hi, Thanks for your review, I have updated as you said :)

**Review by LadyForest — APPROVED** [review]
Thanks for the update, LGTM

**Pull request merged**